clients/java/signalr/src/test/java/WebSocketTransportTest.java

```diff
@@ -2,38 +2,22 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 import com.microsoft.aspnet.signalr.NullLogger;
+import com.microsoft.aspnet.signalr.Transport;
 import com.microsoft.aspnet.signalr.WebSocketTransport;
-import java.net.URISyntaxException;
-import java.util.Arrays;
-import java.util.Collection;
+import org.junit.Rule;
 import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
+import org.junit.rules.ExpectedException;
 
-import static org.junit.Assert.*;
-
-@RunWith(Parameterized.class)
 public class WebSocketTransportTest {
-    private String url;
-    private String expectedUrl;
-
-    public WebSocketTransportTest(String url, String expectedProtocol){
-        this.url = url;
-        this.expectedUrl = expectedProtocol;
-    }
 
-    @Parameterized.Parameters
-    public static Collection protocols(){
-        return Arrays.asList(new String[][] {
-                {"http://example.com", "ws://example.com"},
-                {"https://example.com", "wss://example.com"},
-                {"ws://example.com", "ws://example.com"},
-                {"wss://example.com", "wss://example.com"}});
-    }
+    @Rule
+    public ExpectedException expectedEx = ExpectedException.none();
 
     @Test
-    public void checkWebsocketUrlProtocol() throws URISyntaxException {
-        WebSocketTransport webSocketTransport = new WebSocketTransport(this.url, new NullLogger());
-        assertEquals(this.expectedUrl, webSocketTransport.getUrl().toString());
+    public void WebsocketThrowsIIfItCantConnect() throws Exception {
+        expectedEx.expect(Exception.class);
```

Comment:
We generally prefer using something like `Assert.Throws`. It looks like JUnit 5.x has [`assertThrows`](https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html#assertThrows-java.lang.Class-org.junit.jupiter.api.function.Executable-java.lang.String-) that we could use instead of `ExpectedException`.

Reply:
Do we want to take a new dependency for that though?

Reply:
Isn't it just updating junit to the latest version? We need to register it, but since it's a test-only dependency, I don't think @Eilon will have any concerns. We should be using the latest version anyway.

Reply:
Changing to junit 5 breaks a lot of things and idk if it's worth it atm tbh

Reply:
Ok, but it'll only get harder moving forward :). If it's a lot of work we don't need to do it in this PR.

Reply:
Yeah, I like the JUnit 5 assertThrows syntax more and it also provides a better way to do parametrized tests but I started making the changes and it was just a little to involved to be a side effect of this change imo. I'll file an issue and plan to do this in the preview2 wave.